### ISSUE TITLE
CI builds for PRs

### ISSUE THREAD

**Issue by cleeyv** [ngi-nix/ngipkgs#18]
The ngi0 hydra is configured to build all the packages in ngipkgs: https://hydra.ngi0.nixos.org/jobset/ngipkgs/main#tabs-jobs however, this can only be run for specific branches, not for new PRs, hence the need for a parallel CI solution.

CI has been implemented for PRs using github actions: https://github.com/ngi-nix/ngipkgs/commit/4dfa911c1052e85273ea185bb6a3ea8635d8b1f2
Currently it is running `nix flake check`, which confirms that everything evaluates, and runs all tests, but provides no way to check that all packages build correctly. This is a feature that has been requested more than once over the past few years:
https://github.com/NixOS/nix/issues/3811
https://github.com/NixOS/nix/issues/7165
The first request noted that it would be "very useful in the context of a monorepo". In the comments of the second request, there are some workarounds mentioned, such as adding all packages as "checks" so that they will be built by `nix flake check`, or using `symlinkJoin` to create a single derivation containing all packages which can then be set to the default package for the flake and more easily built in CI: https://www.tweag.io/blog/2022-09-22-rust-nix/

**Comment by cleeyv** [comment]
I tried the symlinkJoin method first but couldn't get it to work. Then I tried adding it to the checks, and once I figured out how to do it, it was very simple: https://github.com/ngi-nix/ngipkgs/commit/99ba38f21f30c0cb5001c826898acd2418a2d848
I got inspiration from this implementation of a build check, which seemed to be also having problems with unwanted darwin builds: https://github.com/ngi-nix/pgp-milter/commit/cae9fd3b825af0dcb71c037570a33e2516681ad0

The result when I pushed the new commit was a CI run that was 8 minutes long (compared to the previous 3 minutes without builds), with logs that included builds of the packages! https://github.com/ngi-nix/ngipkgs/actions/workflows/ci.yaml

The remaining step is testing to confirm that this works as intended with a PR.
